src/lib/installationVerification.ts

```diff
@@ -366,108 +374,96 @@ export class InstallationVerification implements Verifier {
       logger.debug(`retrieveNpmMeta | this.pluginNpmName.scope: ${this.pluginNpmName.scope}`);
       logger.debug(`retrieveNpmMeta | this.pluginNpmName.tag: ${this.pluginNpmName.tag}`);
 
-      if (this.pluginNpmName.scope) {
-        npmRegistry.pathname = path.join(
-          npmRegistry.pathname,
-          `@${this.pluginNpmName.scope}%2f${this.pluginNpmName.name}`
+      const npmShowModule = this.pluginNpmName.scope
+        ? `@${this.pluginNpmName.scope}/${this.pluginNpmName.name}`
+        : this.pluginNpmName.name;
+
+      // run npm show to get metadata
+      const npmCmd = `npm show ${npmShowModule} --registry ${npmRegistry.href} --json`;
+      let npmMetadata: NpmShowResults;
+      try {
+        const npmShowResult = shelljs.exec(npmCmd, { silent: true, fatal: true });
+        npmMetadata = JSON.parse(npmShowResult.stdout);
+      } catch (error) {
+        return reject(new SfdxError(error, 'ShellExecError'));
+      }
+      const meta: NpmMeta = new NpmMeta();
+      meta.moduleName = npmShowModule;
+      logger.debug('retrieveNpmMeta | Found npm meta information.');
+      if (!npmMetadata.versions) {
+        return reject(
+          new SfdxError(
+            `The npm metadata for plugin ${this.pluginNpmName.name} is missing the versions attribute.`,
+            'InvalidNpmMetadata'
+          )
         );
-      } else {
-        npmRegistry.pathname = path.join(npmRegistry.pathname, this.pluginNpmName.name);
       }
-      logger.debug(`retrieveNpmMeta | npmRegistry.pathname: ${npmRegistry.pathname}`);
 
-      this.requestImpl(npmRegistry.href, (err, response, body) => {
-        if (err) {
-          return reject(err);
-        }
-        if (response && response.statusCode === 200) {
-          logger.debug('retrieveNpmMeta | Found npm meta information. Parsing.');
-          const responseObj = JSON.parse(body);
+      // Assume the tag is version tag.
+      let versionNumber = npmMetadata.versions.find((version) => version === this.pluginNpmName.tag);
+
+      logger.debug(`retrieveNpmMeta | versionObject: ${JSON.stringify(versionNumber)}`);
+
+      // If the assumption was not correct the tag must be a non-versioned dist-tag or not specified.
+      if (!versionNumber) {
+        // Assume dist-tag;
+        const distTags = npmMetadata['dist-tags'];
+        logger.debug(`retrieveNpmMeta | distTags: ${JSON.stringify(distTags)}`);
+        if (distTags) {
```

Comment:
can we do away with `get` like
```ts
  const tagVersionStr = distTags[this.pluginNpmName.tag];
```

That function has always felt like a workaround for incomplete TS types and casting all its `unknowns` with `as` sucks.